laser-py/laser.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/python
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# LASER  Language-Agnostic SEntence Representations
+# is a toolkit to calculate multilingual sentence embeddings
+# and to use them for document classification, bitext filtering
+# and mining
+#
+# --------------------------------------------------------
+#
+# Helper functions for tokenization and BPE
+
+import os
+import sys
+import logging
+import sentencepiece as spm
+from sacremoses import MosesPunctNormalizer, MosesDetokenizer
+
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="%(asctime)s | %(levelname)s | %(name)s | %(message)s")
+logger = logging.getLogger("preprocess")
+
+# get environment
+assert os.environ.get('LASER'), 'Please set the enviornment variable LASER'
+
+class LaserTokenizer:          
+    def tokenize(inp_fname, out_fname, spm_model_file, lang='en',
+                lower_case=True, descape=False,
+                verbose=False, over_write=False, gzip=False):
+        assert lower_case, 'lower case is needed by all the models'
+        if not os.path.isfile(out_fname):
+            cat = 'zcat ' if gzip else 'cat '
```

Comment:
To make this more simple, I would suggest perhaps initially working with in-memory data instead of passing around file pointers. For example, we could take as input a list of strings called `texts` and then output a similar datatype (cc/ @avidale).

Reply:
I suggest making the method `tokenize` that works with strings, and also maybe a method `tokenize_file` that reuses `tokenize` and works with files.

Reply:
Suggestion from @avidale sounds good to me. Let's have an option to do both.